hal/network/lwip/cellular/pppncpnetif.cpp

```diff
@@ -151,28 +178,26 @@ void PppNcpNetif::loop(void* arg) {
 
 int PppNcpNetif::up() {
     NetifEvent ev = NetifEvent::Up;
-    int r = os_queue_put(queue_, &ev, 0, nullptr);
-    if (r) {
-        // FIXME: we'll miss this event, just as an attempt to potentially recover
-        // set internal state.
-        up_ = true;
-    }
-    return r;
+    return os_queue_put(queue_, &ev, 0, nullptr);
 }
 
 int PppNcpNetif::down() {
     const auto client = celMan_->ncpClient();
-    if (client->connectionState() != NcpConnectionState::CONNECTED) {
+    /* Note: It's possible that we have powered on the modem but the NCP client initialization is still in progress.
+     * For example, invokes Cellular.connect() followed by Cellular.disconnect().
+     * If we interrupted the operation, we will probably fail to turn off the modem in enable() that is invoked in the PPP thread.
+     * It would result in a hard reset of the modem, which will take longer to turn off the modem. Simply let the NCP client
+     * finalize the initialization procedure and make it reside in ON state. */
+    if (client->connectionState() != NcpConnectionState::CONNECTED && client->ncpState() == NcpState::ON) {
```

Comment:
The idea behind calling disable is to force abort the ongoing (blocking) operation. It may take quite a bit of time for NCP state go to `ON`. I'm not sure it's that good of an idea to add this check.

The hard reset timing concerns only apply to R410, however they are still lower than waiting for the whole intiialization sequence to complete. If you take a look at it you can see that it may take even several minutes in some conditions.